samples/ProjectGrandmaster.Unity/Assets/AudioSystem.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+
+[RequireComponent(typeof(AudioSource))]
+public class AudioSystem : MonoBehaviour
+{
+    [SerializeField]
+    private AudioClip collisionSound;
+    private AudioSource source;
+    private bool isDropped;
+
+    private void Start()
+    {
+        isDropped = false;
+        source = this.GetComponent<AudioSource>();
+    }
+    private void OnTriggerEnter(Collider collision)
+    {
+        if (Time.frameCount < 100) return;
+        source.clip = collisionSound;
```

Comment:
Is the AudioSource exclusive to this component? If so, we could set in this in Start